internal/controllers/machineset/machineset_controller.go

```diff
@@ -571,6 +597,60 @@ func (r *Reconciler) syncMachines(ctx context.Context, s *scope) (ctrl.Result, e
 	return ctrl.Result{}, nil
 }
 
+func newMachineUpToDateCondition(s *scope) (*metav1.Condition, error) {
+	// If the current MachineSet is a stand-alone MachineSet, the MachineSet controller do not set an up-to-data condition
+	// on machines, allowing tools managing higher level abstractions to set this condition.
+	// This is also consistent with the fact that the MachineSet controller primarily takes care of the number of Machine
+	// replicas, it doesn't reconcile them (even if we have a few exceptions like in-place propagation of a few selected
+	// fields and remediation).
+	if s.owningMachineDeployment == nil {
+		return nil, nil
+	}
+
+	// Determine current and desired state.
+	// If the current MachineSet is owned by a MachineDeployment, we mirror the what is implemented in the MachineDeployment controller
+	// to create new trigger rollouts (by creating new MachineSets).
+	// More specifically:
+	// - desired state for the Machine is the spec.Template of the MachineDeployment
+	// - current state for the Machine is the spec.Template of the MachineSet who owns the Machine
+	// Note: We are intentionally considering  current spec from the MachineSet instead of spec from the Machine itself in
+	// order to surface info consistent with what the MachineDeployment controller uses to take decisions about rollouts;
+	// the downside, is that the system will ignore out of band changes applied to controlled Machine, which is
+	// considered an acceptable trade-off given that out of band changes are the exception (user should not change
+	// object owned by the system).
+	// However, In out of band changes happens, at least the system will ignore out of band changes consistently, both in the
+	// MachineDeployment controller and in the condition computed here.
+	current := &s.machineSet.Spec.Template
+	desired := &s.owningMachineDeployment.Spec.Template
+
+	upToDate, _, conditionMessages, err := mdutil.MachineTemplateUpToDate(current, desired)
+	if err != nil {
+		return nil, err
+	}
+
+	if s.owningMachineDeployment != nil && s.owningMachineDeployment.Spec.RolloutAfter != nil {
```

Comment:
```suggestion
	if s.owningMachineDeployment.Spec.RolloutAfter != nil {
```
redundant because of l.606 now